esmvaltool/diag_scripts/quantilebias/quantilebias.R

```diff
@@ -0,0 +1,151 @@
+provenance <- list()
+
+# extract metadata
+models_name <- unname(sapply(metadata, "[[", "dataset"))
+reference_model <- unname(sapply(metadata, "[[", "reference_dataset"))[1]
+models_start_year <- unname(sapply(metadata, "[[", "start_year"))
+models_end_year <- unname(sapply(metadata, "[[", "end_year"))
+models_experiment <- unname(sapply(metadata, "[[", "exp"))
+models_ensemble <- unname(sapply(metadata, "[[", "ensemble"))
+
+ref_idx <- which(models_name == reference_model)
+ref_data_file <- climofiles[ref_idx]
+
+## Loop through input models
+for (model_idx in c(1:(length(models_name)))) {
+  if (model_idx == ref_idx) {
+     next
+  }
+  # Setup parameters and path
+  exp <- models_name[model_idx]
+  year1 <- models_start_year[model_idx]
+  year2 <- models_end_year[model_idx]
+  infile <- climofiles[model_idx]
+  model_exp <- models_experiment[model_idx]
+  model_ens <- models_ensemble[model_idx]
+
+  inregname <- paste0(exp, "_", model_exp, "_", model_ens, "_",
+                      toString(year1), "-", toString(year2), "_", varname)
+  outfile <- paste0(work_dir, "/", inregname, "_", perc_lev, "qb.nc")
+  print(paste0(diag_base, ": pre-processing file: ", infile))
+
+  print(paste0(diag_base, ": ", perc_lev, " percent quantile"))
+  system(paste0("cdo selvar,", varname, " ", infile, " tmp_model.nc"))
+
+  # Remap reference onto model grid
+  system(paste0("cdo remapcon,tmp_model.nc -selvar,", varname, " ",
+                ref_data_file, " tmp_ref.nc"))
+
+  # Get (75)th percentile of reference dataset
+  system(paste0("cdo timpctl,", perc_lev, " tmp_ref.nc  -timmin ",
+                "tmp_ref.nc -timmax tmp_ref.nc tmp_ref_perc_p.nc"))
+
+  # Select points with monthly precipitation greater than (75)th perc
+  system("cdo ge tmp_ref.nc  tmp_ref_perc_p.nc tmp_mask_ref.nc")
+  system("cdo ge tmp_model.nc tmp_ref_perc_p.nc tmp_mask_model.nc")
+
+  # Precipitation sums
+  system("cdo timsum -mul tmp_mask_ref.nc tmp_ref.nc tmp_ref_sum.nc")
+  system("cdo timsum -mul tmp_mask_model.nc tmp_model.nc tmp_model_sum.nc")
+
+  # Calculate quantile bias, set name and attributes
+  system("cdo div tmp_model_sum.nc tmp_ref_sum.nc tmp_qb1.nc")
+  cdo_command <- paste(
+    "cdo ",
+    " -setattribute,qb@standard_name='precipitation_quantile_bias'",
+    " -setattribute,qb@long_name='Precipitation quantile bias'",
+    " -setattribute,qb@units=' '",
+    paste0(" -chname,", varname, ",qb"),
+    " tmp_qb1.nc tmp_qb.nc"
+  )
+  print(cdo_command)
+  system(cdo_command)
+
+  # Copy file to output destination and remove temporary files
+  mv_command <- paste("mv tmp_qb.nc", outfile)
+  print(mv_command)
+  system(mv_command)
+  rm_command <- paste("rm tmp_*")
+  print(rm_command)
+  system(rm_command)
+
+  # Set provenance for this output file
+  caption <- paste0("Quantile bias ", perc_lev, "% for years ",
+                    year1, " to ", year2, " according to ", exp)
+  xbase <- list(list(infile), recipe_docs$authors, recipe_docs$projects,
```

Comment:
Do not copy the authors, project, references from the recipe: in the future someone may write a different recipe that re-uses your diagnostic script, but the recipe is by a different author and made for a different project, etc. Also, esmvaltool already records provenance information from the recipe.